Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -201,6 +201,13 @@ class HTMLFastPathParser {
     {
     }
 
+    HTMLFastPathParser(StringParsingBuffer<CharacterType>&& parsingBuffer, Document& document, ContainerNode& destinationParent)
```

Comment:
You should not need to modify the fast parser at all in this patch.

Reply:
The fastParser only takes a span from the original string, by doing the trimming there we avoid the requirement to create a new substring. No version where the trimming is done before calling the parser will be as optimal.

Reply:
Update tryFastParsingHTMLFragment() to take a `StringView` instead of a `const String`. It shouldn't impact the implementation of the fast parser and it means the call site can create a substring view, which is cheap.